layers/+distributions/spacemacs-base/funcs.el

```diff
@@ -1088,14 +1085,28 @@ Decision is based on `dotspacemacs-line-numbers'."
                (* 1000 (car (spacemacs/mplist-get dotspacemacs-line-numbers
                                                   :size-limit-kb)))))))
 
+;; mode in :enabled, not in :disabled ==> t
+;; mode not in :enabled, in :disabled ==> nil
+;; mode in :enabled, parent in :disabled ==> t
+;; parent in :enabled, mode in :disabled ==> nil
+;; not in :enabled, not in :disabled, :enabled is empty ==> t
+;; not in :enabled, not in :disabled, :enabled is not empty ==> nil
+;; both :enabled and :disabled are empty ==> t
 (defun spacemacs//linum-enabled-for-current-major-mode ()
   "Return non-nil if line number is enabled for current major-mode."
-  (let ((modes (spacemacs/mplist-get dotspacemacs-line-numbers
-                                     :enabled-for-modes)))
-    (memq major-mode modes)))
-
-(defun spacemacs//linum-disabled-for-current-major-mode ()
-  "Return non-nil if line number is disabled for current major-mode."
-  (let ((modes (spacemacs/mplist-get dotspacemacs-line-numbers
-                                     :disabled-for-modes)))
-    (memq major-mode modes)))
+  (let* ((enabled-for-modes (spacemacs/mplist-get dotspacemacs-line-numbers
+                                                  :enabled-for-modes))
+         (disabled-for-modes (spacemacs/mplist-get dotspacemacs-line-numbers
+                                                   :disabled-for-modes))
+         (enabled-for-parent (apply #'derived-mode-p enabled-for-modes))
+         (disabled-for-parent (apply #'derived-mode-p disabled-for-modes)))
+    (or
+     ;; current mode or a parent is in :enabled-for-modes, and there isn't a
+     ;; more specific parent (or the mode itself) in :disabled-for-modes
+     (and enabled-for-parent
+          (not (let ((major-mode enabled-for-parent))
+                 (derived-mode-p disabled-for-parent))))
+     ;; current mode (or parent) not explicitly disabled, and :enabled-for-modes
+     ;; not explicitly specified by user (meaning if it isn't explicitly
+     ;; disabled then it's enabled)
+     (and (null enabled-for-modes) (not disabled-for-parent)))))
```

Comment:
I understand now :open_mouth: 

Well, either a comment stating what you just said, a helper function or just leaving it like that might all be good.

The names like `enabled-for-parent` and `derived-mode-p` misled me at first because I thought they were booleans but actually no.

Anyways, I don't know if such tiny readability detail is worth spending so much energy :smile_cat: